### PR TITLE
Add code coverage on unit tests

### PR DESCRIPTION
**Pull request by JohanLorenzo**
r? @Yoric

### CODE REVIEW

**Comment by JohanLorenzo** [comment]
~~r? @Yoric~~.

The latest rust nightly is not pulled by Travis yet. Clearing review.

**Comment by JohanLorenzo** [comment]
> The latest rust nightly is not pulled by Travis yet. Clearing review.
> Scratch that. Actually rustc `b678600ac 2016-03-29` is published at [2016-03-30](https://static.rust-lang.org/dist/2016-03-30/index.html). I corrected travis.

r? @Yoric 

Note: Please do not merge this PR yet. We need to sync the merge with thinkerbell, foxbox and [rustpi2](https://github.com/fabricedesre/rustpi2).

@fabricedesre It seems we need to [activate coveralls](https://travis-ci.org/fxbox/taxonomy/builds/120015428#L646) on this repo too. I remember you are the one who has the rights, right?

File: src/backend.rs [hunk 1]
```diff
@@ -399,9 +399,9 @@ impl State {
                 // Ensure that we release the borrow before calling `cb`.
                 let borrow = &*service.borrow();
                 let view = ServiceView::new(borrow);
-                matches = selectors.iter().find(|selector| {
+                matches = selectors.iter().any(|selector| {
```
Comment (Yoric) [inline src/backend.rs:416]: Definitely works for me, but why do we need this change? (and all the others of the same kind)

**Comment by Yoric** [comment]
r=me

File: src/backend.rs (hunk 1 above)
Comment (JohanLorenzo) [inline src/backend.rs:416]: That's a new rule coming from the linter. https://github.com/Manishearth/rust-clippy/wiki#search_is_some

File: src/api.rs [hunk 2]
```diff
@@ -3,13 +3,13 @@
 //!
 //! This API is provided as Traits to be implemented:
 //!
-//! - by the low-level layers of the FoxBox, including the adapters;
+//! - by the low-level layers of `FoxBox`, including the adapters;
```
Comment (JohanLorenzo) [inline src/api.rs:6]: New rule of clippy: https://github.com/Manishearth/rust-clippy/wiki#doc_markdown

File: src/lib.rs [hunk 3]
```diff
@@ -1,7 +1,10 @@
 //! This crate defines the high-level API for accessing Connected Devices.
 #![feature(custom_derive, plugin, stmt_expr_attributes)]
 #![plugin(serde_macros)]
+
 #![plugin(clippy)]
+#![deny(clippy)]
```
Comment (JohanLorenzo) [inline src/lib.rs:6]: Compilation now fails if the linter is not happy

File: src/services.rs [hunk 4]
```diff
@@ -376,7 +376,7 @@ impl Parser<ChannelKind> for ChannelKind {
             }
         }
         if source.is_object() {
-            for key in vec!["vendor", "adapter", "kind", "type"] {
+            for key in &["vendor", "adapter", "kind", "type"] {
```
Comment (JohanLorenzo) [inline src/services.rs:379]: New rule of clippy: https://github.com/Manishearth/rust-clippy/wiki#useless_vec

**Comment by fabricedesre** [comment]
I updated to 04-10

**Pull request closed without merging**